contracts/ethregistrar/BulkRenewal.sol

```diff
@@ -3,46 +3,67 @@ pragma experimental ABIEncoderV2;
 
 import "../registry/ENS.sol";
 import "./ETHRegistrarController.sol";
+import "./IETHRegistrarController.sol";
 import "../resolvers/Resolver.sol";
+import "./IBulkRenewal.sol";
 
-contract BulkRenewal {
-    bytes32 constant private ETH_NAMEHASH = 0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
-    bytes4 constant private REGISTRAR_CONTROLLER_ID = 0x018fac06;
-    bytes4 constant private INTERFACE_META_ID = bytes4(keccak256("supportsInterface(bytes4)"));
-    bytes4 constant public BULK_RENEWAL_ID = bytes4(
-        keccak256("rentPrice(string[],uint)") ^
-        keccak256("renewAll(string[],uint")
-    );
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+contract BulkRenewal is IBulkRenewal {
+    bytes32 private constant ETH_NAMEHASH =
+        0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
 
     ENS public ens;
 
-    constructor(ENS _ens) public {
+    constructor(ENS _ens) {
```

Comment:
` ENS public ens;` can be immutable

Reply:
> ```
> ReverseRegistrar: Add new events for etherscan
> ```
> 
> Which events are for etherscan?



> ```
> ReverseRegistrar: Add new events for etherscan
> ```
> 
> Which events are for etherscan?

Actually this was added in a previous PR, it's called `ReverseClaimed`

Reply:
Added!